iothub/device/src/InternalClient.cs

```diff
@@ -953,32 +909,27 @@ internal async Task OnMethodCalledAsync(MethodRequestInternal methodRequestInter
             MethodResponseInternal methodResponseInternal = null;
             byte[] requestData = methodRequestInternal.GetBytes();
 
-            await _methodsDictionarySemaphore.WaitAsync().ConfigureAwait(false);
+            await _methodsSemaphore.WaitAsync().ConfigureAwait(false);
             try
             {
                 Utils.ValidateDataIsEmptyOrJson(requestData);
-                _deviceMethods?.TryGetValue(methodRequestInternal.Name, out callbackContextPair);
-                if (callbackContextPair == null)
+                if (!_deviceMethods.TryGetValue(methodRequestInternal.Name, out callbackContextPair))
```

Comment:
Using `TryGet*` the way it was meant to be used.

Reply:
Can _deviceMethods be null at all? The old code seems to think it can be null. We should add a check for null if it can be null to avoid NRE.

Reply:
I see the explanation for this below. Nevermind.